datasets/ohsumed/ohsumed.py

```diff
@@ -0,0 +1,168 @@
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = "CC BY-NC 4.0"
+
+# TODO: Add link to the official dataset URLs here
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URLs = {"ohsumed": "https://storage.googleapis.com/corona-tweet/ohsumed.zip"}
+
+
+# TODO: Name of the dataset usually match the script name with CamelCase instead of snake_case
+class Ohsumed(datasets.GeneratorBasedBuilder):
+    """OHSUMED: An Interactive Retrieval Evaluation and New Large Test Collection for Research."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    # This is an example of a dataset with multiple configurations.
+    # If you don't want/need to define several sub-sets in your dataset,
+    # just remove the BUILDER_CONFIG_CLASS and the BUILDER_CONFIGS attributes.
+
+    # If you need to make complex sub-parts in the datasets with configurable options
+    # You can create your own builder configuration class to store attribute, inheriting from datasets.BuilderConfig
+    # BUILDER_CONFIG_CLASS = MyBuilderConfig
+
+    # You will be able to load one or the other configurations in the following list with
+    # data = datasets.load_dataset('my_dataset', 'first_domain')
+    # data = datasets.load_dataset('my_dataset', 'second_domain')
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="ohsumed",
+            version=VERSION,
+            description="Config for the entire ohsumed dataset. An Interactive Retrieval Evaluation and New Large Test Collection for Research",
+        )
+    ]
+
+    def _info(self):
+        # TODO: This method specifies the datasets.DatasetInfo object which contains informations and typings for the dataset
+        features = datasets.Features(
+            {
+                "seq_id": datasets.Value("int64"),
+                "medline_ui": datasets.Value("int64"),
+                "mesh_terms": datasets.Value("string"),
+                "title": datasets.Value("string"),
+                "publication_type": datasets.Value("string"),
+                "abstract": datasets.Value("string"),
+                "author": datasets.Value("string"),
+                "source": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
```

Comment:
WOuld be nice to set this since it's a classification task
```python
supervised_keys=(input_feature_name, label_feature_name)
```

Reply:
So I am not certain how to respond to this issue. 
To paraphrase information that's available in the readme files & a preview of the original paper. Link to the published paper is given below. Since its behind a paywall, I could just read the preview. 

https://link.springer.com/chapter/10.1007%2F978-1-4471-2099-5_20

The purpose of the exercise was creating a dataset which could be used for search. Physicians enter the searchterm and some relevant information regarding the patient and then rank the results from the MEDLINE system via their relevance scores (1- possibly relevant; 2- definitely relevant; not relevant topics were not included in the dataset)

The system needs to learn to differentiate between _possibly relevant_ & _definitely relevant_ topics. 

_A document is considered "relevant" to a MeSH "topic" if the MeSH concept name is listed in the MeSH term field (.M) of the document. Please note that the MeSH concepts form a hierarchy. It is common practice to index a document *only* by the most specific MeSH concept that is relevant._

There are 5000+ MeSH terms. Which are similar to something like this- 

"""
Adolescence; Artificial Limbs/EC/UT; Child; Child Health Services/EC/UT; Community Health Services/EC/*UT; Female; Financing, Government; Handicapped/*; Health Services Research; Human; Male; Orthotic Devices/EC/UT; Social Support; Support, Non-U.S. Gov't; Walkers/EC/UT.
"""

topic statement is a query defined by a physician - 

"""
<top>
<num> Number: OHSU1
<title> 60 year old menopausal woman without hormone replacement therapy
<desc> Description:
Are there adverse effects on lipids when progesterone is given with estrogen replacement therapy
</top>
"""


I had deliberately kept it open, by not defining any supervised tasks. But if you want to suggest a different approach, would love to hear about it.

Reply:
If there's not an explicit and favored task that aims at predicting one field given other fields, then we should keep it to None. Here the logic looks a bit different so I don't mind keeping it as None